Source/Core/VideoBackends/D3D/Render.cpp

```diff
@@ -578,14 +578,11 @@ void Renderer::SetViewport()
   Wd = (X + Wd <= GetTargetWidth()) ? Wd : (GetTargetWidth() - X);
   Ht = (Y + Ht <= GetTargetHeight()) ? Ht : (GetTargetHeight() - Y);
 
-  D3D11_VIEWPORT vp = CD3D11_VIEWPORT(
-      X, Y, Wd, Ht,
-      1.0f - MathUtil::Clamp<float>(xfmem.viewport.farZ, 0.0f, 16777215.0f) / 16777216.0f,
-      1.0f -
-          MathUtil::Clamp<float>(xfmem.viewport.farZ - MathUtil::Clamp<float>(xfmem.viewport.zRange,
-                                                                              0.0f, 16777216.0f),
-                                 0.0f, 16777215.0f) /
-              16777216.0f);
+  // We do depth clipping and depth range in the vertex shader instead of relying
+  // on the graphics API. However since we use a 2^24 divisor for all our depth values
+  // to prevent round-trip errors we still need to ensure depth values are clamped
+  // to 2^24 - 1.
+  D3D11_VIEWPORT vp = CD3D11_VIEWPORT(X, Y, Wd, Ht, D3D11_MIN_DEPTH, 16777215.0f / 16777216.0f);
```

Comment:
fwiw given that the whole comment atop this is duplicated multiple times, I would suggest putting the "1677215/16777216" thingy into a named constant instead. That's more semantic than duplicating that constant everywhere, and it allows for moving the comment to the definition of the constant instead of duplicating it everywhere.